packages/syft/src/syft/service/code/user_code.py

```diff
@@ -362,15 +414,76 @@ def user(self) -> UserView | SyftError:
             )
         return api.services.user.get_by_verify_key(self.user_verify_key)
 
+    def _status_from_output_history(
+        self, context: AuthedServiceContext | None = None
+    ) -> UserCodeStatusCollection | SyftError:
+        if context is None:
+            # Clientside
+            api = self._get_api()
+            if isinstance(api, SyftError):
+                return api
+            node_identity = NodeIdentity.from_api(api)
+            is_approved = api.output.has_output_read_permissions(
+                self.id, self.user_verify_key
+            )
+        else:
+            # Serverside
+            node_identity = NodeIdentity.from_node(context.node)
+            output_service = context.node.get_service("outputservice")
+            is_approved = output_service.has_output_read_permissions(
+                context, self.id, self.user_verify_key
+            )
+
+        if isinstance(is_approved, SyftError):
+            return is_approved
+        is_denied = self.l0_deny_reason is not None
+
+        if is_denied:
```

Comment:
probably we want some warning for when we are calling .deny() and you already deposited results